tests/unit/test_network_connections.py

```diff
@@ -3312,6 +3312,114 @@ def test_dns_options_argvalidator(self):
             validator.validate(true_testcase_12)["dns_options"], ["use-vc"]
         )
 
+    def test_dns_argvalidator(self):
+        """
+        Test that configuring ip.dns is not allowed when corresponding IP method for
+        that nameserver is disabled.
+        """
+        input_connection1 = [
```

Comment:
Please use more descriptive names here, what is this testing?

Reply:
I used the descriptive name for connections:
```python
        ipv4_dns_without_ipv4_config = [
            {
                "name": "test_ipv4_dns",
                "type": "ethernet",
                "ip": {
                    "auto6": True,
                    "dhcp4": False,
                    "dns": ["198.51.100.5"],
                },
            }
        ]
        
        ipv6_dns_without_ipv6_config = [
            {
                "name": "test_ipv6_dns",
                "type": "ethernet",
                "ip": {
                    "ipv6_disabled": True,
                    "dhcp4": True,
                    "dns": ["2001:db8::20"],
                },
            }
        ]
```